dsls/sql/src/main/java/org/apache/beam/dsls/sql/rel/BeamAggregationRel.java

```diff
@@ -75,33 +75,36 @@ public BeamAggregationRel(RelOptCluster cluster, RelTraitSet traits
     PCollection<BeamSQLRow> upstream =
         BeamSQLRelUtils.getBeamRelInput(input).buildBeamPipeline(planCreator);
     if (windowFieldIdx != -1) {
-      upstream = upstream.apply("assignEventTimestamp", WithTimestamps
+      upstream = upstream.apply(stageName + "_assignEventTimestamp", WithTimestamps
           .<BeamSQLRow>of(new BeamAggregationTransforms.WindowTimestampFn(windowFieldIdx)));
     }
 
-    PCollection<BeamSQLRow> windowStream = upstream.apply("window",
+    PCollection<BeamSQLRow> windowStream = upstream.apply(stageName + "_window",
         Window.<BeamSQLRow>into(windowFn)
         .triggering(trigger)
         .withAllowedLateness(allowedLatence)
         .accumulatingFiredPanes());
 
     //1. extract fields in group-by key part
-    PCollection<KV<BeamSQLRow, BeamSQLRow>> exGroupByStream = windowStream.apply("exGroupBy",
+    PCollection<KV<BeamSQLRow, BeamSQLRow>> exGroupByStream = windowStream.apply(
+        stageName + "_exGroupBy",
         WithKeys
             .of(new BeamAggregationTransforms.AggregationGroupByKeyFn(windowFieldIdx, groupSet)));
 
     //2. apply a GroupByKey.
     PCollection<KV<BeamSQLRow, Iterable<BeamSQLRow>>> groupedStream = exGroupByStream
-        .apply("groupBy", GroupByKey.<BeamSQLRow, BeamSQLRow>create());
+        .apply(stageName + "_groupBy", GroupByKey.<BeamSQLRow, BeamSQLRow>create());
```

Comment:
Added stageName to all the name because Beam complains that these tranformation does not have stable names. Other Rels might have the similar issue.